mac/Keyman4MacIM/Keyman4MacIM/KMDataRepository.m

```diff
@@ -0,0 +1,185 @@
+/**
+ * Keyman is copyright (C) SIL International. MIT License.
+ *
+ * KMResourcesRepository.m
+ * Keyman
+ *
+ * Created by Shawn Schantz on 2024-07-30.
+ *
+ * Singleton object which serves as an abstraction for the reading and writing of Keyman data.
+ * The 'data' currently consists of keyman keyboards installed by the user. All data is saved locally on disk using NSFileManager.
+ * This is in contrast with the lighter weight Settings which is stored using UserDefaults and handled by KMSettingsRepository.
+ */
+
+#import "KMDataRepository.h"
+#import "KMLogs.h"
+
+@interface KMDataRepository ()
+@property (readonly) NSURL *applicationSupportSubDirectory;   // '~/Library/Application Support'
+@property (readonly) NSURL *documentsSubDirectory;    // '~/Documents'
+@property (readonly) NSURL *obsoleteKeymanKeyboardsDirectory; // '~/Library/Documents/Keyman-Keyboards'
+@end
+
+@implementation KMDataRepository
+
+@synthesize applicationSupportSubDirectory = _applicationSupportSubDirectory;
+@synthesize documentsSubDirectory = _documentsSubDirectory;
+@synthesize keymanKeyboardsDirectory = _keymanKeyboardsDirectory;
+@synthesize obsoleteKeymanKeyboardsDirectory = _obsoleteKeymanKeyboardsDirectory;
+@synthesize keymanDataDirectory = _keymanDataDirectory;
+
+NSString *const kKeyboardsDirectoryName = @"Keyman-Keyboards";
+/* 
+ The name of the subdirectory within '~/Library/Application Support'.
+ The convention is to use bundle identifier.
+ (Using the subsystem id, "com.keyman.app", is a poor choice because the API
+ createDirectoryAtPath sees the .app extension and creates an application file.
```

Comment:
Yes, I'll fix the comments which are incorrect. I prefer `com.keyman.app` as an app ID but since our app ID is already set to `keyman.inputmethod.Keyman` and `com.keyman.app` looks to the file manager like a filename of type `.app`, I plant to stick with `keyman.inputmethod.Keyman` as the name of our app's data directory.